components/system-broker/pkg/http/client.go

```diff
@@ -43,15 +46,9 @@ func NewClient(timeout time.Duration, transport http.RoundTripper) *http.Client
 	}
 }
 
-func NewSecuredHTTPClient(timeout time.Duration, roundTripper HTTPRoundTripper, provider TokenProvider) (*http.Client, error) {
-	transport := &SecuredTransport{
-		roundTripper:  roundTripper,
-		tokenProvider: provider,
-		lock:          sync.Mutex{},
-	}
-
+func NewSecuredHTTPClient(timeout time.Duration, securedTransport HTTPRoundTripper) (*http.Client, error) {
```

Comment:
if we pass the transport from outside, im not sure that NewSecuredHTTPClient describes what the clients purpose is (e.g. you can pass any transport and make it not sercured http client

Reply:
Either we create the client directly and does not use misleading constructor or accept a securedTransport struct directly instead of interface. What do you think?